llvm/lib/Target/X86/AsmParser/X86AsmParser.cpp

```diff
@@ -3784,6 +3795,66 @@ bool X86AsmParser::processInstruction(MCInst &Inst, const OperandVector &Ops) {
     Inst.setOpcode(X86::INT3);
     return true;
   }
+#define FROM_TO(FROM, TO)                                                      \
+  case X86::FROM: {                                                            \
+    if (ForcedOpcodePrefix == OpcodePrefix_EVEX_CMP_TEST) {                    \
```

Comment:
It's strange to introduce ` OpcodePrefix_EVEX_CMP_TEST`. Why not check `X86::isCMP` `X86::isTEST` directly?

Reply:
2dd7c4a